src/applications/personalization/preferences/actions/index.js

```diff
@@ -29,7 +29,10 @@ export const SET_DISMISSED_DASHBOARD_PREFERENCE_BENEFIT_ALERTS =
 
 // load the benefits the user has picked to learn more about
 export function fetchUserSelectedBenefits() {
-  return dispatch => {
+  return (dispatch, getState) => {
+    if (getState().preferences.selectedBenefitsCached) {
+      return null;
```

Comment:
What if we dispatched an action that indicates the cache is being used. I'm not sure if this suggestion is idiomatic redux, though- @jbalboni might have an opinion. The idea is to give feedback via the redux dev tools so it's clear that nothing is broken when the cache is used.

Reply:
I certainly like that idea... but am with you in not knowing if I _should_ add that. I'm leaning towards doing it.

Reply:
I'm not necessarily against it, but I don't think it's idiomatic Redux in that the action would not change the state and would not be handled in a reducer (which might actually throw an error).

Reply:
Are there other instances when an Action will be dispatched but there's no associated Reducer? I don't see a reason to add a Reducer case for handle the `USING_CACHE` action in this case. Is that the potentially non-idiomatic thing you're thinking about?

Reply:
Right, that's what I mean. If you dispatch a `USING_CACHE` action there's nothing for a reducer to do in response because the state isn't changing. An unhandled action normally means something has gone wrong; that's why there are middlewares like https://github.com/socialtables/redux-unhandled-action.

Reply:
heh, posted my comment before seeing yours :) @jbalboni do you have any strong thoughts on the thunk immediately returning `null` in the case that the data is cached? Although it totally works, it somehow feels a little wrong for reasons I can't quite put my finger on.